pymc_experimental/model_builder.py

```diff
@@ -258,6 +254,23 @@ def build_model(
         """
         raise NotImplementedError
 
+    def save_model_coords(self, X: Union[pd.DataFrame, pd.Series], y: pd.Series):
```

Comment:
This should be an abstract method, since it doesn't actually implement anything (the assignment of None, to a property which is originally initialised to None)

Reply:
Yes, this method does not do anything. I don't know exactly how python abstract methods work, but like this I hoped the specific implementation would be optional for those cases where the model coords actually need to be set. For non-hierarchical models, this method does not need to be touched.

Reply:
The reason I didn't even include it here is exactly what you're mentioning here: A lot of models won't use the coords at all, but this here would force the models to contain property that will never be used, along with its setter method. In my opinion is best to not include it at all, and allow for developers of specific child model to include this, as it was done in case of DelayedSaturatedMMM from pymc-marketing